### PR TITLE
PTQ support for OVModelForZeroShotImageClassification

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
# What does this PR do?

Add support for full quantization of zero-shot-image-classification pipelines (OVModelForZeroShotImageClassification ).

| Model ID           | FP32 Accuracy    | INT8 WC Accuracy | INT8 PTQ Accuracy|
|---------------------------------|---------------|--------------------|---------------------|
| openai/clip-vit-base-patch16                |  68.33%       |     68.25%         |      67.40%         | 
| openai/clip-vit-base-patch32                 |   63.37%      |        63.30%      |        61.92%       | 
| openai/clip-vit-large-patch14                  |    75.55%   |     75.52%         |     74.89%          | 

| Model ID           | INT8 WC Text Encoder Speedup    | INT8 WC Vision Encoder Speedup | INT8 PTQ Text Encoder Speedup    | INT8 PTQ Vision Encoder Speedup | 
|---------------------------------|---------------|--------------------|---------------------|------------------|
| openai/clip-vit-base-patch16          236,1188      |     1.42x    |       1.45x       |        2.51x       |        2.35x     |
| openai/clip-vit-base-patch32          222,365       |    1.40x     |     1.61x        |         2.52x      |       2.45x      |
| openai/clip-vit-large-patch14        554,4923          |    1.40x   |       1.62x       |       2.77x        |     2.89x        |

WC stands for weight compression.

INT8 PTQ was applied like this: 
```
optimum-cli export openvino -m openai/clip-vit-base-patch16 --quant-mode int8 \
  --smooth-quant-alpha 0.6 --dataset conceptual_captions ./save_dir
```

Accuracy and speedup were measured using https://github.com/l-bat/CLIP_benchmark with a command like this: 
```
python clip_benchmark/cli.py eval --model_type "optimum_intel" --model openai \
  --pretrained clip-vit-base-patch16 --dataset "imagenet1k" --seed 0 --task "zeroshot_classification" \
  --output "eval.json" --dataset_root <dataset_path> 
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/intel/openvino/modeling.py [hunk 1]
```diff
@@ -996,3 +997,23 @@ def forward(self, input_ids, pixel_values, attention_mask: Optional[torch.Tensor
...
+        )
+
+        if is_data_aware_quantization:
+            from optimum.intel import OVQuantizer
+
+            quantization_config_copy = copy.deepcopy(quantization_config)
+            quantization_config_copy.processor = quantization_config.processor or model_id
+            ov_quantizer = OVQuantizer(model)
+            ov_quantizer.quantize(ov_config=OVConfig(quantization_config=quantization_config_copy))
+        return model
```
Comment (IlyasMoutawwakil) [inline optimum/intel/openvino/modeling.py:1019]: shouldn't this be done in OVBaseModel ? https://github.com/huggingface/optimum-intel/blob/1d8cdcaf241b599369bc3d2e57a0be471db9d5ed/optimum/intel/openvino/modeling_base.py#L399

Comment (nikita-savelyevv) [reply optimum/intel/openvino/modeling.py]: Yes, indeed. Removed.

File: tests/openvino/test_quantization.py [hunk 2]
```diff
@@ -416,7 +456,8 @@ def test_automodel_static_quantization(
     ):
         model_id = MODEL_NAMES[model_name]
         task = model_cls.export_feature
-        dataset_name, dataset_config_name, column_name = _TASK_TO_DATASET[task]
+        dataset_kwargs = {**_TASK_TO_DATASET[task]}
```
Comment (IlyasMoutawwakil) [inline tests/openvino/test_quantization.py:459]: why exploding this dict only to re-create it ?

**Review by IlyasMoutawwakil — APPROVED** [review]
LGTM

File: tests/openvino/test_quantization.py (hunk 2 above)
Comment (nikita-savelyevv) [reply tests/openvino/test_quantization.py:459]: This is because we need `dataset_kwargs` to contain a copy of `_TASK_TO_DATASET[task]` since we modify it below `dataset_kwargs.pop("column_name")`.

**Comment by nikita-savelyevv** [comment]
@echarlaix @IlyasMoutawwakil Are you ok with merging it now so it will be available in the upcoming release?

**Review by echarlaix — APPROVED** [review]
